services/provider/server/server.go

```diff
@@ -334,6 +336,40 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 					"csi.storage.k8s.io/node-stage-secret-name":        nodeCephClientSecret,
 					"csi.storage.k8s.io/controller-expand-secret-name": provisionerCephClientSecret,
 				})})
+		case "CephFilesystemSubVolumeGroup":
+			subVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
+			err := s.client.Get(ctx, types.NamespacedName{Name: i.Name, Namespace: s.namespace}, subVolumeGroup)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get %s cephFilesystemSubVolumeGroup. %v", i.Name, err)
+			}
+			clusterID := fmt.Sprintf("%s-%s-file-%s", subVolumeGroup.Namespace, subVolumeGroup.Spec.FilesystemName, subVolumeGroup.Name)
+
+			nodeCephClientSecret, err := s.getCephClientSecretName(ctx, i.CephClients["node"])
+			if err != nil {
+				return nil, err
+			}
+
+			provisionerCephClientSecret, err := s.getCephClientSecretName(ctx, i.CephClients["provisioner"])
+			if err != nil {
+				return nil, err
+			}
+
+			extR = append(extR, &pb.ExternalResource{
+				Name: "cephfs",
+				Kind: "StorageClass",
+				Data: mustMarshal(map[string]string{
+					"clusterID": hash(clusterID),
```

Comment:
Why are we splitting the creation of the cluster id into two places? (here and Line 345)
Can we just do it here, or there?